plugins/CoreVisualizations/JqplotDataGenerator/Chart.php

```diff
@@ -171,6 +177,14 @@ public function setAxisXLabelsMultiple($xLabels, $seriesToXAxis, $ticks = null)
         }
     }
 
+    /**
+     * @param array<string> $archiveStates
+     */
+    public function setArchiveStates(array $archiveStates): void
+    {
+        $this->archiveStates = $archiveStates;
```

Comment:
Related to comment about making it clear these are directly linked to data points, does it make sense to make this defensive and prevent someone passing 4 archive states for 3 data points for example?

Reply:
Added a "consistency" check to log if the series or states have differing amounts of entries.

Reply:
Toned down the logs quite a bit. I missed the expected state that, when for example comparison "November" with "October", it is expected to have different amount of series points.

So only logging an info message if the state information not span the full graph ("longest series" should match state information count).